overlord/snapstate/snapstate.go

```diff
@@ -337,6 +343,25 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		ts.MarkEdge(checkAsserts, DownloadAndChecksDoneEdge)
 	}
 
+	if preseedMode {
+		if installHook == nil {
```

Comment:
maybe we should add a panic in MarkEdge itself if the task arg is nil

Reply:
That's probably a good idea. I've removed the check here. I'll make a followup for just MarkEdge change.